tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -12620,21 +12620,39 @@ HLSLWaveSizeAttr *ValidateWaveSizeAttributes(Sema &S, Decl *D,
                                              const AttributeList &A) {
   // validate that the wavesize argument is a power of 2 between 4 and 128
   // inclusive
-  HLSLWaveSizeAttr *pAttr = ::new (S.Context)
-      HLSLWaveSizeAttr(A.getRange(), S.Context, ValidateAttributeIntArg(S, A),
-                       A.getAttributeSpellingListIndex());
+  HLSLWaveSizeAttr *pAttr = ::new (S.Context) HLSLWaveSizeAttr(
```

Comment:
We will need to emit an error if the WaveSize attribute has more than one argument and the shader model target is less than 6.8.

Reply:
We don't need to do this now.  Besides, it must be checked based on active entry points.